src/treq/auth.py

```diff
@@ -68,18 +413,22 @@ def add_basic_auth(
     :returns: :class:`~twisted.web.iweb.IAgent`
     """
     if not isinstance(username, bytes):
-        username = username.encode('utf-8')
+        username = username.encode("utf-8")
     if not isinstance(password, bytes):
-        password = password.encode('utf-8')
+        password = password.encode("utf-8")
 
-    creds = binascii.b2a_base64(b'%s:%s' % (username, password)).rstrip(b'\n')
+    creds = binascii.b2a_base64(b"%s:%s" % (username, password)).rstrip(b"\n")
     return _RequestHeaderSetterAgent(
         agent,
-        Headers({b'Authorization': [b'Basic ' + creds]}),
+        Headers({b"Authorization": [b"Basic " + creds]}),
     )
 
 
-def add_auth(agent, auth_config):
+def add_digest_auth(agent: IAgent, http_digest_auth: HTTPDigestAuth) -> IAgent:
+    return _RequestDigestAuthenticationAgent(agent, http_digest_auth)
```

Comment:
I don't think this should be public, as `add_auth` appears to be the "front door" for this API, and this has no docstring, so let's give it an underscore.